gtsam/linear/AcceleratedPowerMethod.h

```diff
@@ -0,0 +1,133 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010-2019, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   AcceleratedPowerMethod.h
+ * @date   Sept 2020
+ * @author Jing Wu
+ * @brief  accelerated power method for fast eigenvalue and eigenvector
+ * computation
+ */
+
+#pragma once
+
+// #include <gtsam/base/Matrix.h>
+// #include <gtsam/base/Vector.h>
+#include <gtsam/linear/PowerMethod.h>
+
+namespace gtsam {
+
+using Sparse = Eigen::SparseMatrix<double>;
+
+/* ************************************************************************* */
+/// MINIMUM EIGENVALUE COMPUTATIONS
+
+// Template argument Operator just needs multiplication operator
+template <class Operator>
+class AcceleratedPowerMethod : public PowerMethod<Operator> {
+  /**
+   * \brief Compute maximum Eigenpair with power method
+   *
+   * References :
+   * 1) Rosen, D. and Carlone, L., 2017, September. Computational
+   * enhancements for certifiably correct SLAM. In Proceedings of the
+   * International Conference on Intelligent Robots and Systems.
+   * 2) Yulun Tian and Kasra Khosoussi and David M. Rosen and Jonathan P. How,
+   * 2020, Aug, Distributed Certifiably Correct Pose-Graph Optimization, Arxiv
+   * 3) C. de Sa, B. He, I. Mitliagkas, C. Ré, and P. Xu, “Accelerated
+   * stochastic power iteration,” in Proc. Mach. Learn. Res., no. 84, 2018, pp.
+   * 58–67
+   *
+   * It performs the following iteration: \f$ x_{k+1} = A * x_k + \beta *
+   * x_{k-1} \f$ where A is the certificate matrix, x is the Ritz vector
+   *
+   */
+
+  double beta_ = 0;  // a Polyak momentum term
+
+  Vector previousVector_;  // store previous vector
+
+ public:
+  // Constructor from aim matrix A (given as Matrix or Sparse), optional intial
+  // vector as ritzVector
+  explicit AcceleratedPowerMethod(
+      const Operator &A, const boost::optional<Vector> initial = boost::none)
+      : PowerMethod<Operator>(A, initial) {
+    Vector x0;
+    // initialize ritz vector
+    x0 = initial ? Vector::Random(this->dim_) : initial.get();
+    Vector x00 = Vector::Random(this->dim_);
+    x0.normalize();
+    x00.normalize();
+
+    // initialize Ritz eigen vector and previous vector
+    previousVector_ = update(x0, x00, beta_);
+    this->ritzVector_ = update(previousVector_, x0, beta_);
+    this->perturb();
+
+    // set beta
+    Vector init_resid = this->ritzVector_;
+    const double up = init_resid.transpose() * this->A_ * init_resid;
+    const double down = init_resid.transpose().dot(init_resid);
+    const double mu = up / down;
+    beta_ = mu * mu / 4;
+    setBeta();
```

Comment:
1. Added beta argument in the constructor so that user can directly set the beta value. If there is no default value provide by the user, then it will call the `estimateBeta()` 
2. I don't understand why Shonan will already have a good idea about what beta should be. My understanding is the optimal beta will  be the {\lamda_2}^2/4 of the aim matrix, did we get the approximate value in previous power method? If yes, is it the  ritz value associated with the maximum eigenvector, because we assume here \lamda_1 is close to \lamda_2?